docs/sdg/sdg-api-interface.md

```diff
@@ -0,0 +1,64 @@
+# SDG Library Design
+
+## Objective
+
+Library called `instructlab-sdg` that can be called per seed example that includes question and answer pairs, and context for grounded skills.
+
+## Structure of the SDG Library
+
+We propose the following structure for the SDG library. There will be config files that contain all the prompt templates for the pipelines.
+
+```markdown
+- src/instructlab/sdg/
+    - configs/ 
+        - gen_q.yaml 
+        - gen_a.yaml
+        - ...
+    - init.py 
+    - block.py 
+    - llmblock.py 
+    - pipeline.py 
+    - sdg.py
+```
+
+## CLI
+
+The CLI client uses the instructlab SDG library and provides it a run configuration with input parameters.
+
+```python
+# cli_driver.py
```

Comment:
I interpreted this as sample code for what `ilab` code might look like.

or did you envision something included in the `sdg` repository? The filename here made me stop and think I may have assumed wrong

Reply:
yes that is correct sample code for the CLI or the driver side